corehq/ex-submodules/couchexport/models.py

```diff
@@ -141,12 +141,14 @@ def last(cls, index):
 
     @classmethod
     def get_all_checkpoints(cls, index):
-        return cls.view("couchexport/schema_checkpoints",
+        doc_ids = [result["id"] for result in cls.get_db().view(
+            "couchexport/schema_checkpoints",
             startkey=[json.dumps(index)],
             endkey=[json.dumps(index), {}],
-            include_docs=True,
             reduce=False,
-        )
+        )]
+        for doc in iter_docs(cls.get_db(), doc_ids):
+            yield cls.wrap(doc)
```

Comment:
It's loading all docs into memory at some point, but not all at the same time. Because of `iter_docs` (whose default chunksize is 100), only 100 docs will be in memory at a given time. That's the important difference.